ldms/src/ldmsd/ldmsd_request.c

```diff
@@ -774,36 +773,41 @@ static int string2attr_list(char *str, struct attr_value_list **__av_list,
 	return rc;
 }
 
-int validate_ldmsd_req(ldmsd_req_ctxt_t reqc, ldmsd_req_hdr_t rh)
+int validate_ldmsd_req(ldmsd_req_hdr_t rh)
 {
 	ldmsd_req_attr_t attr;
 	unsigned int off;
+	uint32_t rec_len = ntohl(rh->rec_len);
+	uint32_t attr_len;
+
 	/* Request consists of only the header and terminating discriminator */
-	if (rh->rec_len == sizeof(*rh) + sizeof(attr->discrim)) {
+	if (rec_len == sizeof(*rh) + sizeof(attr->discrim)) {
 		attr = (ldmsd_req_attr_t)(rh + 1);
-		if (attr->discrim != 0)
+		if (ntohl(attr->discrim) != 0)
 			return 0;
 		else
 			return 1;
 	}
 	/* Request contains, at least, one attribute*/
-	else if (rh->rec_len < sizeof(*rh) + sizeof(*attr))
+	else if (rec_len < sizeof(*rh) + sizeof(*attr))
 		return 0;
 
 	off = sizeof(*rh);
 	attr = (ldmsd_req_attr_t)(rh+1);
 
-	while (attr->discrim) {
-		off += sizeof(*attr) + attr->attr_len;
+	while (ntohl(attr->discrim)) {
+		attr_len = ntohl(attr->attr_len);
+
+		off += sizeof(*attr) + attr_len;
 		/* Check attr doesn't reference beyond end of request buffer */
-		if (off + sizeof(attr->discrim) > rh->rec_len)
+		if (off + sizeof(attr->discrim) > rec_len)
 			return 0;
 		/* Check null termination of each attr_value */
-		else if (attr->attr_value[attr->attr_len-1] != '\0')
+		else if (attr->attr_value[attr_len-1] != '\0')
```

Comment:
This will break if attr_len is zero. it should be something like:
```
else if (attr_len && attr->attr_value[attr_len-1] != '\0')
```

Reply:
Thanks for catching this. I just pushed the change.